### PR TITLE
minimum/maximum builtins

### PR DESCRIPTION
**Pull request by Snektron**
This pull request adds `@minimum` and `@maximum` builtin functions. The main use case for these is SIMD (See #903), though they work on scalars as well.

For floats, this uses the `llvm.maxnum` and `llvm.minnum` intrinsics. Note that x86_64 `vminps` and `vmaxps` have different semantics, and so these generate some extra instructions:
```
c5 f0 5d d0              vminps xmm2,xmm1,xmm0
c5 f8 c2 c0 03           vcmpunordps xmm0,xmm0,xmm0
c4 e3 69 4a c1 00        vblendvps xmm0,xmm2,xmm1,xmm0
c3  
  ```
  
  On other architectures (riscv and arm) this shouldn't be a problem.

### CODE REVIEW

**Comment by Snektron** [comment]
Oh, and i also made sure to implement the float semantics in comptime: if either argument is nan, the other is returned.

**Comment by Snektron** [comment]
CI errors seem unrelated

File: src/stage1/ir.cpp [hunk 1]
```diff
@@ -3289,14 +3289,115 @@ static void float_mod(ZigValue *out_val, ZigValue *op1, ZigValue *op2) {
...
+}
+
 static void float_negate(ZigValue *out_val, ZigValue *op) {
     out_val->type = op->type;
     if (op->type->id == ZigTypeIdComptimeFloat) {
         bigfloat_negate(&out_val->data.x_bigfloat, &op->data.x_bigfloat);
     } else if (op->type->id == ZigTypeIdFloat) {
         switch (op->type->data.floating.bit_count) {
             case 16:
-                out_val->data.x_f16 = f16_neg(op->data.x_f16);
```
Comment (nektro) [inline src/stage1/ir.cpp:3299]: why was this line removed?

Comment (Snektron) [reply src/stage1/ir.cpp:3299]: That must have been an accident. Thanks for yhe find :)
